### ISSUE TITLE
RFXCOM: Message DATE_TIME not implemented

### ISSUE THREAD

**Issue by bhelm** [openhab/openhab#1626]
I get the following error with RFXCOM RFXtrx433E on Arm with openhab 1.6.0 snapshot from yesterday (first 1.6.0 version i tried).

```
10:30:36.197 DEBUG o.o.b.r.i.RFXComConnection[:152] - Unknown data received, data: 0D580105B9000301020507270489
10:30:36.215 ERROR o.o.b.r.internal.RFXComBinding[:279] - Error occured during packet receiving, data: 0D580105B9000301020507270489
org.openhab.binding.rfxcom.internal.RFXComException: Message DATE_TIME not implemented
        at org.openhab.binding.rfxcom.internal.messages.RFXComMessageFactory.getMessageInterface(RFXComMessageFactory.java:121)
        at org.openhab.binding.rfxcom.internal.RFXComBinding$MessageLister.packetReceived(RFXComBinding.java:231)
        at org.openhab.binding.rfxcom.internal.connector.RFXComSerialConnector$SerialReader.run(RFXComSerialConnector.java:202)
Caused by: java.lang.ClassNotFoundException: org.openhab.binding.rfxcom.internal.messages.RFXComDateTimeMessage
        at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:501)
        at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:421)
        at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:412)
        at org.eclipse.osgi.internal.baseadaptor.DefaultClassLoader.loadClass(DefaultClassLoader.java:107)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Class.java:259)
        at org.openhab.binding.rfxcom.internal.messages.RFXComMessageFactory.getMessageInterface(RFXComMessageFactory.java:116)
        ... 2 common frames omitted
10:30:36.477 DEBUG o.o.b.r.i.RFXComConnection[:152] - Unknown data received, data: 0D580106B9000301020507270489
10:30:36.502 ERROR o.o.b.r.internal.RFXComBinding[:279] - Error occured during packet receiving, data: 0D580106B9000301020507270489
org.openhab.binding.rfxcom.internal.RFXComException: Message DATE_TIME not implemented
        at org.openhab.binding.rfxcom.internal.messages.RFXComMessageFactory.getMessageInterface(RFXComMessageFactory.java:121)
        at org.openhab.binding.rfxcom.internal.RFXComBinding$MessageLister.packetReceived(RFXComBinding.java:231)
        at org.openhab.binding.rfxcom.internal.connector.RFXComSerialConnector$SerialReader.run(RFXComSerialConnector.java:202)
Caused by: java.lang.ClassNotFoundException: org.openhab.binding.rfxcom.internal.messages.RFXComDateTimeMessage
        at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:501)
        at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:421)
        at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:412)
        at org.eclipse.osgi.internal.baseadaptor.DefaultClassLoader.loadClass(DefaultClassLoader.java:107)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Class.java:259)
        at org.openhab.binding.rfxcom.internal.messages.RFXComMessageFactory.getMessageInterface(RFXComMessageFactory.java:116)
        ... 2 common frames omitted
```
```
java version "1.8.0_06"
Java(TM) SE Runtime Environment (build 1.8.0_06-b23)
Java HotSpot(TM) Client VM (build 25.6-b23, mixed mode)
```

**Comment by bhelm** [comment]
Still valid for 1.6.2:
```
22:12:42.134 ERROR o.o.b.r.internal.RFXComBinding[:279] - Error occured during packet receiving, data: 0D580116B90003041D030D150A69
org.openhab.binding.rfxcom.internal.RFXComException: Message DATE_TIME not implemented
        at org.openhab.binding.rfxcom.internal.messages.RFXComMessageFactory.getMessageInterface(RFXComMessageFactory.java:121)
Caused by: java.lang.ClassNotFoundException: org.openhab.binding.rfxcom.internal.messages.RFXComDateTimeMessage
        at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:501)
22:12:42.397 ERROR o.o.b.r.internal.RFXComBinding[:279] - Error occured during packet receiving, data: 0D580117B90003041D030D150A69
org.openhab.binding.rfxcom.internal.RFXComException: Message DATE_TIME not implemented
        at org.openhab.binding.rfxcom.internal.messages.RFXComMessageFactory.getMessageInterface(RFXComMessageFactory.java:121)
Caused by: java.lang.ClassNotFoundException: org.openhab.binding.rfxcom.internal.messages.RFXComDateTimeMessage
        at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:501)
```

**Comment by paulianttila** [comment]
@bhelm: As you see from the error, support for DATE_TIME message is not implemented yet. Support will be available only if someone will implemented it and make a pull request. Could you do it by yourself?

@teichsta: bug label can be removed.

**Comment by federic0** [comment]
Is there a way to "silence" the logs for these messages?

**Comment by Speederc** [comment]
Hi @bhelm,
I tried to generate a new version with this DATE_TIME support : http://dr-petelot-marion.chirurgiens-dentistes.fr/temp/org.openhab.binding.rfxcom_1.9.0.201601221259.jar .
It should work with the following item declarations :
DateTime dateTime {rfxcom=">SensorId:DateTime"}
Can you test it and give me your feedback ?
Thanks

**Comment by 9037568** [comment]
@speederc, do you plan to submit a PR for your changes?

**Comment by Speederc** [comment]
@9037568 , the PR #3956 was already submitted for this message type. Unfortunately it wasn't tested nor validated in this thread.

**Comment by 9037568** [comment]
Closed via merged PR #3956.
